pkg/initializer/core/core.go

```diff
@@ -0,0 +1,148 @@
+)
+
+var supportedExtensions = []string{"js", "java", "py", "sh"}
+
+var languageForFileExtension = map[string]string{
+	"sh"	:  	"shell",
+	"java"	: 	"java",
+	"js"	:   "node",
+	"py"	: 	"python",
+}
+
+type Initializer struct {
+	Initialize func(options.InitOptions) error
+}
+
+func Initialize(language string, ext string, opts options.InitOptions) error {
+	functionPath, err := ResolveFunctionPath(opts, ext)
+	if err != nil {
+		return err
+	}
+
+	if opts.Artifact != "" && languageForFileExtension[filepath.Ext(opts.Artifact)[1:]] != language {
+		return errors.New(fmt.Sprintf("language %s conflicts with artifact file extension %s",language, opts.Artifact))
+	}
+
+
+	// Create function resources in function Path
+	opts.FunctionName, _ = functions.FunctionNameFromPath(opts.FunctionPath)
+
+	if opts.Input == "" {
+		opts.Input = opts.FunctionName
+	}
+
+	if opts.Artifact =="" {
+		opts.Artifact = filepath.Base(functionPath)
+	}
+
+	var protocolForLanguage = map[string]string{
+		"shell"	:  	"stdio",
+		"java"	: 	"http",
+		"js"	:   "http",
+		"node"	:   "http",
+		"py"	: 	"stdio",
+	}
+
+	if opts.Protocol == "" {
+		opts.Protocol = protocolForLanguage[language]
+	}
+
+	workdir := filepath.Dir(functionPath)
+
+	err = generate.CreateFunction(workdir,language, opts)
+	return err
+}
+
+
+//Assumes given file paths have been sanity checked and are valid
+func ResolveFunctionPath(options options.InitOptions, ext string) (string, error) {
+
+
+	absFilePath, err := filepath.Abs(options.FunctionPath)
+	if err != nil {
+		return "", err
+	}
+
+	var resolvedFunctionPath string
+	var functionDir string
+	var functionFile string
+	if osutils.IsDirectory(absFilePath) {
+		if options.Artifact == "" {
+			functionFile = options.FunctionName
+			functionDir = absFilePath
+			if ext != "" {
+				resolvedFunctionPath = filepath.Join(functionDir, fmt.Sprintf("%s.%s", functionFile, ext))
+			} else {
+				functionFile, err = searchForFunctionResource(functionDir, options.FunctionName)
+				if err != nil {
+					return "", err
+				}
+				resolvedFunctionPath = functionFile
+			}
+		} else {
+			resolvedFunctionPath = filepath.Join(absFilePath, options.Artifact)
+		}
+	} else {
+		resolvedFunctionPath = absFilePath
+	}
+	if !osutils.FileExists(resolvedFunctionPath) {
+		return "", errors.New(fmt.Sprintf("function path %s does not exist", resolvedFunctionPath))
+	}
+	return resolvedFunctionPath, nil
+
+}
+
+func searchForFunctionResource(dir string, functionName string) (string, error) {
+	files, err := filepath.Glob(filepath.Join(dir, "*"))
+	if err != nil {
+		return "", err
+	}
+
+	foundFile := ""
+	for _, f := range (files) {
+		if b := filepath.Base(f); b[0:len(b)-len(filepath.Ext(f))] == functionName {
+			for _, e := range supportedExtensions {
+				if filepath.Ext(f) == "."+e {
+					if foundFile == "" {
+						foundFile = f
+					} else {
+						return "", errors.New(fmt.Sprintf("function file is not unique %s, %s", filepath.Base(foundFile), filepath.Base(f)))
```

Comment:
What if more than two functions are indicated? Perhaps foundFile could be a slice and the error message could warn about all the conflicting "function files". Just a suggestion - you may not think this is worth the extra complexity.